### PR TITLE
feat(guix_shell): Initial implementation

### LINKED ISSUE #3999: Guix shell support
#### Is your feature request related to a problem? Please describe.

No support for Guix environments. I became reliant on this prompt for Nix shells so why not for Guix shells, too. :)

#### Describe the solution you'd like

There is a way to detect whether one is inside of a Guix shell through the `GUIX_ENVIRONMENT` variable. For more details, see the [related page](https://guix.gnu.org/en/manual/devel/en/html_node/Invoking-guix-shell.html#Invoking-guix-shell).

#### Describe alternatives you've considered

No alternative solution found so far.

### PR DESCRIPTION
**Pull request by delafthi**
#### Description
Added guix_shell module to display a symbol when a guix shell is active.

#### Motivation and Context
It makes it a lot easier to see if a current guix shell is active. Additionally, the feature is requested in #3999 

Closes #3999

#### Screenshot
guix_shell

#### How Has This Been Tested?
- [x] I have tested with `cargo test guix_shell`
- [x] I have tested with `starship module guix_shell` without an active guix shell environment
- [x] I have tested with `starship module guix_shell` with an active guix shell environment

- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:
- [x] I have updated the documentation accordingly (only english).
- [x] I have updated the tests accordingly.

#### Additional discussion:
As seen in the screenshot the icon does not represent the Guix logo. However, this is the icon I get when I copy the icon from the [Nerd fonts cheat sheet](https://www.nerdfonts.com/cheat-sheet) (search for guix). The icon does not seem to exist in v2.2.2, at least not in Victor Mono (the font I use).

### CODE REVIEW

**Review by davidkna — CHANGES_REQUESTED** [review]
Please replace the symbol, but the rest looks good to me.

File: docs/config/README.md [hunk 1]
```diff
@@ -2451,6 +2452,39 @@ style = "yellow"
...
+
+The `guix_shell` module shows the [guix-shell](https://guix.gnu.org/manual/devel/en/html_node/Invoking-guix-shell.html) environment.
+The module will be shown when inside a guix-shell environment.
+
+### Options
+
+| Option     | Default                    | Description                                            |
+| ---------- | -------------------------- | ------------------------------------------------------ |
+| `format`   | `'via [$symbol]($style) '` | The format for the module.                             |
+| `symbol`   | `" "`                     | A format string representing the symbol of guix-shell. |
```
[the remark this answers is not in the document]
Comment (delafthi) [reply docs/config/README.md:2465]: I'm currently in holidays. I'll make the changes as soon as I'm back (in approx. 1.5 weeks)

**Comment by delafthi** [comment]
Sry, for the little force push mess I created. Squashed a fixup into the wrong (initial commit) and some other stuff. Now the initial commit should be back to where it was before, and the changes you reqested in [5ba870a](https://github.com/starship/starship/pull/4397/commits/5ba870a1a9c695d1f1323c81588fddbd8c5f5d76) and [7276982](https://github.com/starship/starship/pull/4397/commits/7276982e3dba3a7aa6d0836ffbd2d26fcf8928d3). In [097d210](https://github.com/starship/starship/pull/4397/commits/097d21032e61e60664605cd5fe02accf1803df82) I moved the guix shell config documentation to where it belongs in the alphabetical order.

**Comment by delafthi** [comment]
Sry for the force-push mess I created. I squashed a commit into the initial feature commit, and had some other back and forth until everything worked. Now the initial commit should be back to where it was before and all new changes in the last three commits.

**Review by davidkna — APPROVED** [review]

**Review by andytom — APPROVED** [review]
LGTM

**Comment by andytom** [comment]
Thank you for your contribution @delafthi.

**Pull request merged**